.github/workflows/govulncheck.yml

```diff
@@ -0,0 +1,100 @@
+# This workflow is meant to run govulncheck on all the branches
+# that are containing a maintained version of OpenTofu.
+# For more considerations about this, check this PR: https://github.com/opentofu/opentofu/pull/2600
+#
+# This will attempt to create an issue with the results for each scanned version
+# in case any vulnerability is found. If the issue for the scanned branch already exists
+# the run will just print the scan output and will leave the issue untouched.
+#
+# This is meant to run _only_ from the main branch, on a scheduled manner.
+# All the other branches will be scanned directly by the run triggered from the main branch.
+
+name: Govulncheck
+
+on:
+  schedule:
+    - cron: '42 3 * * SUN'
+  workflow_dispatch: {}
+
+jobs:
+  gather-branches:
+    name: Collect the branches that needs to be scanned
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    env:
+      GH_TOKEN: ${{ github.token }}
+    outputs:
+      branches: ${{ steps.branches.outputs.branches }}
+    steps:
+      - id: branches
+        run: |
+          ## This is using GitHub API and we manually filter out the versions that are not supported anymore
+          # gh api "repos/opentofu/opentofu/branches?protected=true" --paginate | tee out 
+          # cat out | jq '.[] | select(.name != "v1.6") | [.name]' | jq -sc 'add' | tee branches
+          # echo "branches=$(cat branches)">> "$GITHUB_OUTPUT"
+          
+          ## This is using https://endoflife.date since there 1.6 is already marked as it reached EOL
+          ## This is adding manually the main branch to scan it as well
+          curl https://endoflife.date/api/opentofu.json | tee out 
+          cat out | jq '.[] | select(.eol == false) | [.cycle]' | jq -sc 'add' | jq -c '["main"] + .' | tee branches
+          echo "branches=$(cat branches)">> "$GITHUB_OUTPUT"
```

Comment:
This second approach is using https://endoflife.date/opentofu and is nice in terms that we don't really need to maintain this workflow anymore, once merged.
Pros:
* no maintenance needed

Cons:
* adds dependency on a 3rd party system

Reply:
We are keeping only the hardcoded strategy to hold the control on what branches we run the flow against.
See [this](https://github.com/opentofu/opentofu/pull/2636#discussion_r2033011305) suggestion.